Add win32.aarch64 target environment to eclipse-platform-parent/pom.xml

HannesWell opened the pull request:
This is (probably) required to support building the Eclipse-SDK on Windows for ARM.

See https://github.com/eclipse-platform/eclipse.platform.swt/pull/1045#discussion_r1529883121

HannesWell commented:
Looks like we need a launcher for win32.aarch64 first.

laeubi commented:
> Looks like we need a launcher for win32.aarch64 first.

Yes that would be the next logical step, @chirontt would you mind to open a PR against equinox to bring in the launcher parts?

laeubi commented:
btw something looks strange here:

Could not resolve plugin org.eclipse.equinox.launcher.win32.win32.aarch64_null;

chirontt commented:
> > Looks like we need a launcher for win32.aarch64 first.
> 
> Yes that would be the next logical step, @chirontt would you mind to open a PR against equinox to bring in the launcher parts?

eclipse-equinox/equinox#559

chirontt commented:
> > Could not resolve plugin org.eclipse.equinox.launcher.win32.win32.aarch64_null;
> 
> looks like there is no version and then Tycho prints it as `null`

It's too early for this submit - all submodules (well, 5 of them) should have `win32.aarch64` support first, before this aggregator repo can begin the `win32.aarch64` support.

I think I will have to create a TODO list of items for all the various PRs for various repos, here in this aggregator as a central space. How should I go about with this TODO list? via an `Issue`? or via `Discussion`? or via a `Pull Request`?

laeubi commented:
I think you can use this issue here:
- https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/577

you can simply edit your inital posting and add a todo list.

HannesWell commented:
> @HannesWell as [discussed](https://github.com/eclipse-platform/eclipse.platform.ui/pull/1787#issuecomment-2030305378), here's my branch to merge with this PR, thanks:
> 
> https://github.com/chirontt/eclipse.platform.releng.aggregator/tree/add-win32.aarch64-environment

I have now added your commit to this PR but removed the addition of the `org.eclipse.core.resources.win32.aarch64` and `org.eclipse.core.filesystem.win32.aarch64` fragments since they are not (yet) available and adjusted the commit message since your commit is then mainly about publishing the new artifacts and products.

Unfortunately this build currently fails due to https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2014.

@iloveeclipse FYI, without this there will probably be comparator errors again, similar to https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2011 since the Equinox binaries were re-compiled, but this change touches `rcp.config`.
Actually I wanted to submit this today to avoid such error, but https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2014 is blocking that unfortunately.
